pkg/resource/deploy/source_eval.go

```diff
@@ -2453,9 +2453,13 @@ func (rm *resmon) RegisterResource(ctx context.Context,
 	replaceOnChanges := opts.ReplaceOnChanges
 
 	replacementTrigger := resource.NewNullProperty()
-	if req.GetReplacementTrigger() != nil {
+	replacementTriggerValue := opts.GetReplacementTrigger()
+	if replacementTriggerValue == nil {
+		replacementTriggerValue = req.GetReplacementTrigger()
+	}
```

Comment:
How can this be on req but not on opts? That looks like we'd re-add the trigger if a transform deleted it?